src/super_gradients/training/models/pose_estimation_models/yolo_nas_pose.py

```diff
@@ -0,0 +1,239 @@
+import collections
+import copy
+from functools import lru_cache
+from typing import Union, Tuple, Optional, List
+
+import numpy as np
+import torch.nn.functional
+from omegaconf import DictConfig
+from torch import nn, Tensor
+
+from super_gradients.common.decorators.factory_decorator import resolve_param
+from super_gradients.common.factories.activations_type_factory import ActivationsTypeFactory
+from super_gradients.common.factories.processing_factory import ProcessingFactory
+from super_gradients.common.object_names import Models
+from super_gradients.common.registry import register_model, register_detection_module
+from super_gradients.modules import BaseDetectionModule, ConvBNAct
+from super_gradients.training.models.arch_params_factory import get_arch_params
+from super_gradients.training.models.detection_models.customizable_detector import CustomizableDetector
+from super_gradients.training.pipelines.pipelines import PoseEstimationPipeline
+from super_gradients.training.processing.processing import Processing
+from super_gradients.training.utils import HpmStruct, get_param, DEKRPoseEstimationDecodeCallback
+from super_gradients.training.utils.media.image import ImageSource
+from super_gradients.training.utils.predict import ImagesPoseEstimationPrediction
+
+
+@register_detection_module()
+class YoloNASHead(BaseDetectionModule):
+    @resolve_param("activation_type", ActivationsTypeFactory())
+    def __init__(
+        self,
+        num_classes: int,
+        in_channels: Tuple[int, int, int],
+        inter_channels: int,
+        channels_per_kpt: int,
+        num_blocks: int,
+        activation_type,
+        upsample_factor: int,
+    ):
+        super().__init__(in_channels)
+        self.num_classes = num_classes
+        self.num_classes_with_center = num_classes + 1
+        self.channels_per_kpt = channels_per_kpt
+        self.stem = ConvBNAct(sum(in_channels), inter_channels, kernel_size=1, stride=1, padding=0, bias=False, activation_type=activation_type)
+        self.heatmap = self.build_heatmap_path(inter_channels, activation_type, num_blocks, self.num_classes_with_center, upsample_factor)
+
+        self.offset_transition = ConvBNAct(
+            inter_channels, channels_per_kpt * num_classes, kernel_size=3, stride=1, padding=1, bias=False, activation_type=activation_type
+        )
+
+        offset_modules = []
+        for _ in range(num_classes):
+            offset_modules.append(self.build_offset_path(inter_channels, activation_type, num_blocks, channels_per_kpt, upsample_factor))
```

Comment:
If I am not mistaken, this can be implemented more efficiently by using 
Grouped Convolution (where groups=num_classes). It will save slice and concat but also can be optimized by the GPU much better. 

![image](https://github.com/Deci-AI/super-gradients/assets/4509334/0bc1fb52-62f2-49c6-ad74-0ed28dc100c9)

Reply:
This looks like a good idea from inference time perspective, however we need to empiricaly test for this change affects mAP. In the [paper](https://arxiv.org/pdf/2104.02300.pdf) authors discuss this briefly in section `3.4. Discussions`. Sadly they do not provide numbers for group-conv approach.

Reply:
Just to leave some traces for future generations - we did try the group conv approach. 
Yes, ONNX looked much better, but in fact we gained zero improvement of the throughput.